include/mercury/kes_dashboard_live_tiles_cli/IDownloadable.h

```diff
@@ -0,0 +1,24 @@
+/**
+ * \file IDownloadable.h
+ *
+ * \copyright Copyright (C) 2019 Kano Computing Ltd.
+ *            License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
+ *
+ * \brief     TODO
+ */
+
+
+#ifndef __KES_DASHBOARD_LIVE_TILES_CLI_I_DOWNLOADABLE_H__
+#define __KES_DASHBOARD_LIVE_TILES_CLI_I_DOWNLOADABLE_H__
+
+
+#include <parson.h>
+
+
+class IDownloadable {
```

Comment:
Abstract base classes need to [always define a virtual destructor](https://www.studytonight.com/cpp/virtual-destructors.php):

```c++
virtual ~IDownloadable {}
```

It is up to you if you make the destructor pure virtual or not, my preference is to simply have it virtual and defined so that interface implementors don't need to explicitly define a destructor if they don't need one.

Reply:
Yikes! Will definitely add that in, thanks! 👍